openpype/lib/abstract_load_template.py

```diff
@@ -0,0 +1,305 @@
+            return
+
+        current_asset = avalon.io.Session["AVALON_ASSET"]
+        self.populate_template(current_asset,
+                               loaders_by_name,
+                               placeholder_class)
+
+    @property
+    def template_path(self):
+        """
+        Property returning template path. Avoiding setter.
+        Getting template path from open pype settings based on current avalon
+        session and solving the path variables if needed.
+
+        Returns:
+            str: Solved template path
+
+        Raises:
+            ValueError: No profile found from settings for current avalon
+                session
+            KeyError: Could not solve path because a key does not exists
+                in avalon context
+            IOError: Solved path does not exists on current filesystem
+        """
+        project = avalon.io.Session["AVALON_PROJECT"]
+        anatomy = Anatomy(project)
+        project_settings = get_project_settings(project)
+        current_dcc = avalon.io.Session["AVALON_APP"]
+        current_task = avalon.io.Session["AVALON_TASK"]
+        build_info = project_settings[current_dcc]['templated_workfile_build']
+        profiles = build_info['profiles']
+
+        for profile in profiles:
+            if current_task in profile['task_types']:
+                path = profile['path']
+                break
+        else:
+            raise ValueError(
+                "No matching profile found for task '{}' in DCC '{}'".format(
+                    current_task, current_dcc)
+            )
+
+        try:
+            solved_path = None
+            while True:
+                solved_path = anatomy.path_remapper(path)
+                if solved_path == path:
+                    break
+                path = solved_path
+        except KeyError as missing_key:
+            raise KeyError(
+                "Could not solve key '{}' in template path '{}'".format(
+                    missing_key, path))
+
+        if not os.path.exists(solved_path):
+            raise IOError(
+                "Template found in openPype settings for task '{}' with DCC \
+                '{}' does not exists. (Not found : {})".format(
+                    current_task, current_dcc, solved_path))
+        return solved_path
+
+    def populate_template(self, current_asset,
+                          loaders_by_name, placeholder_class):
+        """
+        Use template placeholders to load assets and parent them in hierarchy
+
+        Arguments :
+            current_asset (str):
+            loader_by_name (dict(name:loader)):
+            placeholder_class (AbstractPlaceHolder):
+
+        Returns:
+            None
+        """
+        current_asset_entity = avalon.io.find_one({
+            "type": "asset",
+            "name": current_asset
+        })
+
+        linked_asset_entities = get_linked_assets(current_asset_entity)
```

Comment:
Rather use variable name containing `docs` than `entities`. Entity is usually more complex object than a data holder.